### PR TITLE
Consul: agent config updates for WI

### PR DESCRIPTION
**Pull request by tgross**
This changeset makes two changes:
* Removes the `consul.use_identity` field from the agent configuration. This behavior is properly covered by the presence of `consul.service_identity` / `consul.task_identity` blocks.
* Adds a `consul.task_auth_method` and `consul.service_auth_method` fields to the agent configuration. This allows the cluster administrator to choose specific Consul Auth Method names for their environment, with a reasonable default.

Docs updates will happen in https://github.com/hashicorp/nomad/pull/18685, where I'm doing the rest of the documentation.

### CODE REVIEW

**Review by lgfa29 — APPROVED** [review]

File: client/allocrunner/consul_hook.go [hunk 1]
```diff
@@ -79,14 +68,14 @@ func (h *consulHook) Prerun() error {
...
+	if err := h.prepareConsulTokensForServices(tg.Services, tokens, tg); err != nil {
 		mErr.Errors = append(mErr.Errors, err)
 	}
 	for _, task := range tg.Tasks {
-		if err := h.prepareConsulTokensForServices(task.Services, tokens); err != nil {
+		if err := h.prepareConsulTokensForServices(task.Services, tokens, tg); err != nil {
 			mErr.Errors = append(mErr.Errors, err)
 		}
-		if err := h.prepareConsulTokensForTask(job, task, tg.Name, tokens); err != nil {
+		if err := h.prepareConsulTokensForTask(job, task, tg, tokens); err != nil {
```
[the remark this answers is not in the document]
Comment (pkazmierczak) [reply client/allocrunner/consul_hook.go:78]: It is not, I have [a draft PR](https://github.com/hashicorp/nomad/pull/18780) that fixes it

**Review by pkazmierczak — APPROVED** [review]

File: nomad/structs/structs.go [hunk 2]
```diff
@@ -8190,6 +8190,16 @@ func (t *Task) Warnings() error {
...
 }
 
+func (t *Task) GetConsulClusterName(tg *TaskGroup) string {
+	if t.Consul != nil && t.Consul.Cluster != "" {
+		return t.Consul.Cluster
+	}
+	if tg != nil && tg.Consul != nil && tg.Consul.Cluster != "" {
+		return tg.Consul.Cluster
+	}
+	return ConsulDefaultCluster
```
[the remark this answers is not in the document]
Comment (tgross) [reply nomad/structs/structs.go:8200]: Oh shoot, I totally forgot about the namespace controls on this. Lemme take this back to the drawing board.

File: client/allocrunner/consul_hook.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgross) [reply client/allocrunner/consul_hook.go]: These two PRs will conflict in the merge anyways, so I've done that here as well.

File: nomad/structs/structs.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tgross) [reply nomad/structs/structs.go]: Removed from this code entirely and I'll open a ENT PR for that.

**Pull request merged**